askup/views.py

```diff
@@ -242,6 +246,112 @@ def user_profile_view(request, user_id):
     )
 
 
+def sign_up_view(request):
+    """
+    Provide a sign up view.
+    """
+    if request.method == 'POST':
+        form = SignUpForm(request.POST)
+        redirect = validate_sign_up_form_and_create_user(form, request)
+
+        if redirect:
+            return redirect
+    else:
+        form = SignUpForm
+
+    return render(request, 'askup/sign_up_form.html', {'form': form})
+
+
+def validate_sign_up_form_and_create_user(form, request):
+    """
+    Validate the Sign Up form and create a user on success.
+
+    Creates a user and emails the activation url to him/her on success.
+    """
+    if form.is_valid():
+        user = form.save(commit=False)
+        user.is_active = False
+        user.save()
+        send_user_an_activation_email(user, request)
+        email = user.email
+        user.groups = [Group.objects.get(name='Student')]
+        domain = Domain.objects.filter(name=email[email.find('@') + 1:]).first()
+        add_organizations_to_user(
+            (form.cleaned_data['organization'], (domain and domain.organization)),
+            user
+        )
+
+        return redirect('askup:sign_up_activation_sent')
+
+    return None
+
+
+def add_organizations_to_user(organizations, user):
+    """
+    Iterate over organizations tuple and add the non-empty organizations to the user.
+    """
+    for organization in organizations:
+        if organization:
```

Comment:
why?

Reply:
Cause it may hold a None value as an organization.
now did this loop with set({}) instead of tuple()

Reply:
👍